pkg/apis/core/v1/conversion.go

```diff
@@ -539,3 +539,50 @@ func Convert_core_PersistentVolumeSpec_To_v1_PersistentVolumeSpec(in *core.Persi
 func Convert_v1_PersistentVolumeSpec_To_core_PersistentVolumeSpec(in *v1.PersistentVolumeSpec, out *core.PersistentVolumeSpec, s conversion.Scope) error {
 	return autoConvert_v1_PersistentVolumeSpec_To_core_PersistentVolumeSpec(in, out, s)
 }
+
+// Convert_v1_ServiceSpec_To_core_ServiceSpec converts service spec from v1 to core
+func Convert_v1_ServiceSpec_To_core_ServiceSpec(in *v1.ServiceSpec, out *core.ServiceSpec, s conversion.Scope) error {
+	// conver the automatic part
+	if err := autoConvert_v1_ServiceSpec_To_core_ServiceSpec(in, out, s); err != nil {
+		return err
+	}
+
+	// cluster ip plural in core, single and plural in v1.
+	// If both fields (v1.ClusterIPs and v1.ClusterIP) are provided and differ, then ClusterIP is authoritative for compatibility with older kubelets
+	if (len(in.ClusterIP) > 0 && len(in.ClusterIPs) > 0) && (in.ClusterIP != in.ClusterIPs[0]) && out.Type != core.ServiceTypeExternalName {
```

Comment:
@khenidak would it be clearer to reorg this function like:

```
if out.Type == core.ServiceTypeExternalName {
    if len(out.ClusterIPs) == 1 && out.ClusterIPs[0] == "" {
    }
} else {
    if (len(in.ClusterIP) > 0 && len(in.ClusterIPs) > 0) && (in.ClusterIP != in.ClusterIPs[0]) {
    }
    if len(in.ClusterIP) > 0 && len(in.ClusterIPs) == 0 {
    }
}
```

Reply:
conversion has been removed entirely. and this issue has been resolved.